app/views/directives/key-value-editor.html

```diff
@@ -0,0 +1,239 @@
+            type="text"
+            class="form-control"
+            ng-class="{ '{{setFocusValClass}}' : $last  }"
+            id="{{uniqueForValue(unique, $index)}}"
+            name="{{uniqueForValue(unique, $index)}}"
+            ng-attr-placeholder="{{ (!isReadonlyAny) && valuePlaceholder || ''}}"
+            ng-minlength="{{valueMinlength}}"
+            maxlength="{{valueMaxlength}}"
+            ng-model="entry.value"
+            ng-readonly="isReadonlyAny || isReadonlySome(entry.name) || entry.isReadonly"
+            ng-pattern="validation.val"
+            ng-required="!allowEmptyKeys && entry.value">
+        </div>
+
+        <div ng-if="entry.valueFrom">
+          <div ng-if="isReadonlyAny || entry.isReadonlyValue" class="faux-input-group">
+            <span
+              class="faux-form-control-addon {{entry.valueIcon}}"
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.valueIconTooltip || valueIconTooltip}}"
+              title="{{entry.valueIconTooltip || valueIconTooltip}}"></span>
+            <div
+              class="faux-form-control readonly">
+              <span ng-switch="entry.refType">
+                <span ng-switch-when="configMapKeyRef">
+                  Set to the key {{entry.valueFrom.configMapKeyRef.key}} in config map
+                  <span
+                    ng-if="entry.isReadonlyValue">
+                    {{entry.valueFrom.configMapKeyRef.name}}
+                  </span>
+                  <a
+                    ng-if="!(entry.isReadonlyValue)"
+                    ng-href="{{entry.apiObj | navigateResourceURL}}">
+                    {{entry.valueFrom.configMapKeyRef.name}}
+                  </a>
+                </span>
+                <span ng-switch-when="secretKeyRef">
+                  Set to the key {{entry.valueFrom.secretKeyRef.key}} in secret
+                  <span
+                    ng-if="entry.isReadonlyValue">
+                    {{entry.valueFrom.secretKeyRef.name}}
+                  </span>
+                  <a
+                    ng-if="!(entry.isReadonlyValue)"
+                    ng-href="{{entry.apiObj | navigateResourceURL}}">
+                    {{entry.valueFrom.secretKeyRef.name}}
+                  </a>
+                </span>
+                <span ng-switch-when="fieldRef">
+                  {{entry.valueAlt}}
+                </span>
+                <span ng-switch-default>
+                  {{entry.valueAlt}}
+                </span>
+              </span>
+            </div>
+          </div>
+          <div ng-if="(!isReadonlyAny) && (!entry.isReadonlyValue)">
+            <div class="ui-select">
+              <ui-select ng-model="entry.selectedValueFrom" ng-required="true" on-select="valueFromObjectSelected(entry, $select.selected)">
+                <ui-select-match placeholder="Select config map or secret">
```

Comment:
we made the link configurable to support other options down the road, but then hardcoded this one into the editor.  We made the list of objects generic to support other types, would this be better off as just "Select a resource" or something.